web-client/src/presenter/actions/FileDocument/setDocumentFileForAutoGeneratedEntryOfAppearanceAction.test.ts

```diff
@@ -0,0 +1,47 @@
+import { presenter } from '../../presenter-mock';
+import { runAction } from '@web-client/presenter/test.cerebral';
+import { setDocumentFileForAutoGeneratedEntryOfAppearanceAction } from './setDocumentFileForAutoGeneratedEntryOfAppearanceAction';
+
+describe('setDefaultFileDocumentFormValuesAction', () => {
+  it('should set isUploading to true and not change the primaryDocumentFile if the pdfPreviewUrl is not set', async () => {
+    const result = await runAction(
+      setDocumentFileForAutoGeneratedEntryOfAppearanceAction,
+      {
+        modules: { presenter },
+        state: {
+          form: { primaryDocumentFile: 'this value should not be modified' },
+          pdfPreviewUrl: undefined,
+        },
+      },
+    );
+
+    expect(result.state.form.primaryDocumentFile).toEqual(
+      'this value should not be modified',
+    );
+    expect(result.state.fileUploadProgress.isUploading).toEqual(true);
+  });
+
+  it('should set isUploading to true and modify the primaryDocumentFile when the pdfPreviewUrl is set', async () => {
+    (global as any).fetch = async () => {
+      return {
+        blob: async () => {
+          return new Blob([], { type: 'pdf' });
```

Comment:
check these linting warnings for the test. is there a reason these functions are async?

Reply:
I think we were trying to mimic the behavior of fetch closely- updated the test to use jest mock functions